idea/src/org/jetbrains/kotlin/idea/inspections/RedundantCompanionReferenceInspection.kt

```diff
@@ -65,6 +66,9 @@ class RedundantCompanionReferenceInspection : AbstractKotlinInspection() {
 
             val containingClass = objectDeclaration.containingClass() ?: return false
             if (reference.containingClass() != containingClass && reference == parent.receiverExpression) return false
+            val calleeText = ((selectorExpression as? KtCallExpression) ?: (grandParent as? KtQualifiedExpression)?.callExpression)
```

Comment:
First: this function is already too long, you can't add more logic into it. You should extract 3 lines of code you added into separate function + split code of existing `isRedundantCompanionReference` into subfunctions

Second: `values` and `valueOf` are hardcoded strings. There are already present in the project. At least you can find them in `RemoveRedundantQualifierNameInspection` and `UnusedSymbolInspection`. Please, extract those hardcoded strings into util class/file and reuse them